sashimi-webapp/src/database/storage.js

```diff
@@ -0,0 +1,290 @@
+        .then(success => resolve(true))
+        .catch(sqlError => reject(sqlError));
+      } else {
+        resolve('false');
+      }
+    });
+  } else {
+    throw new exceptions.PromiseFunctionNotDefined();
+  }
+}
+
+function createOrganizationTable(isOrganizationTableFirstInitialize) {
+  if (typeof Promise === 'function') {
+    return new Promise((resolve, reject) => {
+      if (!isOrganizationTableFirstInitialize) {
+        entitiesCreator.createOrganizationTable()
+        .then(success => resolve(true))
+        .catch(sqlError => reject(sqlError));
+      } else {
+        resolve('false');
+      }
+    });
+  } else {
+    throw new exceptions.PromiseFunctionNotDefined();
+  }
+}
+
+function createFolderTable(isFolderTableFirstInitialize) {
+  if (typeof Promise === 'function') {
+    return new Promise((resolve, reject) => {
+      if (!isFolderTableFirstInitialize) {
+        entitiesCreator.createFolderTable()
+        .then(success => resolve(true))
+        .catch(sqlError => reject(sqlError));
+      } else {
+        resolve('false');
+      }
+    });
+  } else {
+    throw new exceptions.PromiseFunctionNotDefined();
+  }
+}
+
+function createFileManagerTable(isFileTableFirstInitialize) {
+  if (typeof Promise === 'function') {
+    return new Promise((resolve, reject) => {
+      if (!isFileTableFirstInitialize) {
+        entitiesCreator.createFileManagerTable()
+        .then(success => resolve(true))
+        .catch(sqlError => reject(sqlError));
+      } else {
+        resolve('false');
+      }
+    });
+  } else {
+    throw new exceptions.PromiseFunctionNotDefined();
+  }
+}
+
+function creationOfTables() {
+  if (typeof Promise === 'function') {
+    return new Promise((resolve, reject) => {
+      initCreateTable()
+      .then(() =>
+        query.isTableExistsInDatabase(constants.ENTITIES_USER)
+        .then(isUserTableFirstInitialize =>
+          createUserTable(isUserTableFirstInitialize)
+          .then(isSuccess => isSuccess)
+          .catch(sqlError => reject(sqlError))
+          )
+        .catch(sqlError => reject(sqlError))
+      )
+      .then(() =>
+        query.isTableExistsInDatabase(constants.ENTITIES_ORGANIZATION)
+        .then(isOrganizationTableFirstInitialize =>
+          createOrganizationTable(isOrganizationTableFirstInitialize))
+          .then(isSuccess => isSuccess)
+          .catch(sqlError => reject(sqlError))
+        .catch(sqlError => reject(sqlError))
+      )
+      .then(() => query.isTableExistsInDatabase(constants.ENTITIES_FOLDER)
+        .then(isFolderTableFirstInitialize =>
+          createFolderTable(isFolderTableFirstInitialize))
+          .then(isSuccess => isSuccess)
+          .catch(sqlError => reject(sqlError))
+        .catch(sqlError => reject(sqlError))
+      )
+      .then(() => query.isTableExistsInDatabase(constants.ENTITIES_FILE_MANAGER)
+        .then(isFileTableFirstInitialize =>
+          createFileManagerTable(isFileTableFirstInitialize)
```

Comment:
You mentioned to me that is the table has already been initialized, then the code will continue to execute the next table initialization. It will be useful is this behaviour is documented.